src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -277,6 +277,11 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
     markRef(current, workInProgress);
 
     if (!shouldUpdate) {
+      // Context providers should defer to sCU for rendering
+      if (hasContext) {
+        invalidateContextProvider(workInProgress, false);
```

Comment:
This change shouldn't be relevant for functional components, since it's specific to context-providers. Functional components can only _consume_ context, since there's no underlying instance. Right?